Tools/AP_Periph/can.cpp

```diff
@@ -1620,15 +870,122 @@ void AP_Periph_FW::can_start()
         }
     }
 #endif
+}
 
-    canardInit(&dronecan.canard, (uint8_t *)dronecan.canard_memory_pool, sizeof(dronecan.canard_memory_pool),
-               onTransferReceived_trampoline, shouldAcceptTransfer_trampoline, this);
+AP_Periph_DroneCAN::AP_Periph_DroneCAN()
+{
+    canard_iface.init(canard_memory_pool, sizeof(canard_memory_pool), PreferredNodeID);
 
-    if (PreferredNodeID != CANARD_BROADCAST_NODE_ID) {
-        canardSetLocalNodeID(&dronecan.canard, PreferredNodeID);
-    }
-}
+    // setup message timeouts and priorities
+    dynamic_node_id_pub.set_priority(CANARD_TRANSFER_PRIORITY_MEDIUM);
+    dynamic_node_id_pub.set_timeout_ms(500);
+
+    node_status_pub.set_priority(CANARD_TRANSFER_PRIORITY_LOW);
+    node_status_pub.set_timeout_ms(1000);
+
+#if HAL_ENABLE_SENDING_STATS
+    stats_pub.set_priority(CANARD_TRANSFER_PRIORITY_LOW);
+    stats_pub.set_timeout_ms(1000);
+
+    can_stats_pub.set_priority(CANARD_TRANSFER_PRIORITY_LOW);
+    can_stats_pub.set_timeout_ms(1000);
+#endif
+
+    mag_pub.set_priority(CANARD_TRANSFER_PRIORITY_MEDIUM);
+    mag_pub.set_timeout_ms(10);
+
+#ifdef HAL_PERIPH_ENABLE_GPS
+    gnss_fix2_pub.set_priority(CANARD_TRANSFER_PRIORITY_MEDIUM);
```

Comment:
we should be able to set a default priority, avoiding all these set_priority() calls

Reply:
Actually, I think it might be a good idea to set the priorities at definition through constructors. Forcing users to set priority should be necessary, so we don't end up with wrong priorities. Wrong set of priorities can cause serious issues during flight time. The same I think goes for timeouts as well.